extensions/binding/org.eclipse.smarthome.binding.ntp/src/main/java/org/eclipse/smarthome/binding/ntp/handler/NtpHandler.java

```diff
@@ -92,29 +92,33 @@ public void handleCommand(ChannelUID channelUID, Command command) {
     public void initialize() {
 
         try {
-            logger.debug("Initializing NTP handler for '{}'.", getThing().getUID());
+            logger.debug("Initializing NTP handler for '{}'.", getThing().getUID().toString());
 
             Configuration config = getThing().getConfiguration();
-            this.hostname = (String) config.get(PROPERTY_NTP_SERVER);
-            this.refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
-            this.refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
-            this.refreshNtpCount = 0;
+            hostname = (String) config.get(PROPERTY_NTP_SERVER);
+            refreshInterval = (BigDecimal) config.get(PROPERTY_REFRESH_INTERVAL);
+            refreshNtp = (BigDecimal) config.get(PROPERTY_REFRESH_NTP);
+            refreshNtpCount = 0;
 
             try {
                 timeZone = TimeZone.getTimeZone((String) config.get(PROPERTY_TIMEZONE));
             } catch (Exception e) {
                 timeZone = TimeZone.getDefault();
-                logger.debug("using default TZ: {}", timeZone);
+                logger.debug("{} using default TZ: {}", getThing().getUID().toString(), timeZone);
             }
 
             try {
                 String localeString = (String) config.get(PROPERTY_LOCALE);
                 locale = new Locale(localeString);
             } catch (Exception e) {
                 locale = Locale.getDefault();
-                logger.debug("using default locale: {}", locale);
+                logger.debug("{} using default locale: {}", getThing().getUID().toString(), locale);
             }
             dateTimeChannelUID = new ChannelUID(getThing().getUID(), CHANNEL_DATE_TIME);
+            updateStatus(ThingStatus.OFFLINE);

```

Comment:
@kai

From the discussion on the forum & documentation, seems that the status
needs to be set in during the initialisation to ONLINE or OFFLINE. Hence I
added it.
see also
https://www.eclipse.org/smarthome/documentation/development/bindings/thing-handler.html

outside of following that approach for the working of the binding there is
no need for it.

```

If the initialization is done, the status must be set to ONLINE resp.
OFFLINE (cf. Thing Status
<https://www.eclipse.org/smarthome/documentation/concepts/things.html#thing-status>).
Only Things in status ONLINE or OFFLINE are considered as *initialized*.

If the Thing is not initializable the configuration can be updated via
ThingHandler.handleConfigurationUpdate(Map). The binding has to notify the
ThingManager about the updated configuration. The ThingManager tries to
initialize the ThingHandler resp. Thingagain.

After the handler is initialized (initialize() has been called, Thing has
status ONLINE / OFFLINE), the handler must be ready to handle methods calls
like handleCommand and handleUpdate, as well as thingUpdated
```

2016-04-13 9:28 GMT+02:00 Kai Kreuzer <notifications@github.com>:

> In
> extensions/binding/org.eclipse.smarthome.binding.ntp/src/main/java/org/eclipse/smarthome/binding/ntp/handler/NtpHandler.java
> <https://github.com/eclipse/smarthome/pull/1377#discussion_r59503864>:
>
> >              }
> >              dateTimeChannelUID = new ChannelUID(getThing().getUID(), CHANNEL_DATE_TIME);
> > +            updateStatus(ThingStatus.OFFLINE);
>
> do you need to go OFFLINE here or is it enough that the automatic refresh
> determines whether it is ONLINE or OFFLINE afterwards?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/smarthome/pull/1377/files/50820acbea8a4f3be77492a0bc87e90fe211379c#r59503864>
>